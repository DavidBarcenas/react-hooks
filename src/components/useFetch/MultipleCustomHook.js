import React from 'react'
import { useFetch } from '../../hooks/useFetch'
import { useCounter } from '../../hooks/useCounter'

export const MultipleCustomHook = () => {
  const { counter, increment } = useCounter(1);
  const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`)
  const { author, quote } = !!data && data[0];

  return (
    <div className="text-center p-5">
      <h1 className="text-primary">useFetch - BreakingBad</h1>
      <hr/>
      
      {
        loading ? (
          <div className="text-center">
            <div className="spinner-border" role="status">
              <span className="sr-only">Loading...</span>
            </div>
          </div>
        ) : (
          <>
            <blockquote className="blockquote m-1">
              <p className="mb-0">{ quote }</p>
              <footer className="blockquote-footer">{ author }</footer>
            </blockquote>

            <button className="btn btn-success mt-4" onClick={increment}>Next</button>
          </>
        )
      }
    </div>
  )
}
