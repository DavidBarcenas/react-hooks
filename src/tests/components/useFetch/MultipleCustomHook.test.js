import '@testing-library/jest-dom';
import '@testing-library/jest-dom';
import React from 'react'
import { shallow } from 'enzyme';
import { MultipleCustomHook } from '../../../components/useFetch/MultipleCustomHook';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')


describe('<MultipleCustomHook /> testing', () => {

  useCounter.mockReturnValue({
    counter: 10
  })
  
  test('should render correctly', () => {

    useFetch.mockReturnValue({
      data: null, loading: true, error: null
    })

    const wrapper = shallow(<MultipleCustomHook />)
    expect(wrapper).toMatchSnapshot()
  })

  
  test('should show info', () => {
    useFetch.mockReturnValue({
      data: [{
         author: 'Daveepro', 
         quote: 'Hello World'
      }], 
      loading: false, 
      error: null
    })

    const wrapper = shallow(<MultipleCustomHook />)
    expect(wrapper.find('.spinner-border').exists()).toBe(false)
  })
  
  

}) 
