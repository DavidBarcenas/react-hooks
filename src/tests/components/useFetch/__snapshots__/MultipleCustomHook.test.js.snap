// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleCustomHook /> testing should render correctly 1`] = `
<div
  className="text-center p-5"
>
  <h1
    className="text-primary"
  >
    useFetch - BreakingBad
  </h1>
  <hr />
  <div
    className="text-center"
  >
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
